### PR TITLE
Better error message for out of stack in assembly.

### PR DESCRIPTION
**Pull request by chriseth**
This specific exception will later also be used to check if code is compilable or not.

### CODE REVIEW

File: libsolidity/codegen/AsmCodeGen.cpp [hunk 1]
```diff
@@ -184,14 +184,26 @@ void CodeGenerator::assemble(
...
+			false,
+			_identifierAccess,
+			_useNamedLabelsForFunctions
+		)(_parsedData);
+	}
+	catch (StackTooDeepError const& _e)
+	{
+		string const* message = boost::get_error_info<errinfo_comment>(_e);
+		BOOST_THROW_EXCEPTION(
+			CompilerError() << errinfo_comment(
```
Comment (axic) [inline libsolidity/codegen/AsmCodeGen.cpp]: `CompilerError` is used for errors in "compiler parameters". I think this should be `InternalCompilerError`?

File: libsolidity/codegen/AsmCodeGen.cpp [hunk 2]
```diff
@@ -184,14 +184,26 @@ void CodeGenerator::assemble(
...
+			*EVMDialect::strictAssemblyForEVM(),
+			_optimize,
+			false,
+			_identifierAccess,
+			_useNamedLabelsForFunctions
+		)(_parsedData);
+	}
+	catch (StackTooDeepError const& _e)
+	{
+		string const* message = boost::get_error_info<errinfo_comment>(_e);
```
Comment (axic) [inline libsolidity/codegen/AsmCodeGen.cpp]: We have `_e.comment()` for this?

**Review by ekpyron — APPROVED** [review]
We don't really have the infrastructure for testing this set up, have we? Anyways, it looks fine to me.

**Comment by chriseth** [comment]
It will be used in an upcoming PR.

**Pull request merged**